### PR TITLE
fix(profiling): The priority thread metadata is optional

### PR DESCRIPTION
**Pull request by Zylphrex**
The priority in the thread metadata is now an optional value. This is causing
failures in validation and rejected profiles.

### CODE REVIEW

**Review by indragiek — APPROVED** [review]

**Review by jan-auer — APPROVED** [review]
Note that clients sending this to old Sentry instances will still cause failures. If you'd like to prevent this, consider marking more fields as optional that aren't absolutely needed for your processing.

I'm approving this to unblock, but please do not deploy this change until #1394 has been merged.

**Review by untitaker — APPROVED** [review]

**Pull request merged**
